platform/commonUI/edit/bundle.js

```diff
@@ -203,20 +205,30 @@ define([
                     ]
                 },
                 {
+                    "key": "save-and-stop-editing",
+                    "category": "save",
+                    "implementation": SaveAndStopEditingAction,
+                    "name": "Save and Done Editing",
```

Comment:
Save and Finish Editing sounds like you'll go back and "finish" editing. Save and Done is awkward, admittedly, but I think is clearer and still brief.